FRC-Extension/RoboRIOCode/RoboRIOConnection.cs

```
@@ -26,73 +30,189 @@ public enum ConnectionUser
+        private ConnectionType m_connectionType = ConnectionType.None;
+
+        private ConnectionInfo m_adminConnectionInfo;
+        private ConnectionInfo m_lvUserConnectionInfo;
+
+        private SshClient m_sshAdminClient;
+        private SshClient m_sshUserClient;
+        private ScpClient m_scpUserClient;
+        private ScpClient m_scpAdminClient;
 
-        public static ConnectionInfo AdminConnectionInfo => s_adminConnectionInfo;
-        public static ConnectionInfo LvUserConnectionInfo => s_lvUserConnectionInfo;
+        public IPAddress IPAddress => m_remoteIp;
 
-        public static async Task<ConnectionReturn> CheckConnectionAsync(string teamNumberS)
+        public ConnectionStatus ConnectionStatus
         {
-            return await CheckConnectionAsync(teamNumberS, TimeSpan.FromSeconds(2)).ConfigureAwait(false);
+            get
+            {
+                if (m_remoteIp == null)
+                {
+                    return null;
+                }
+                return new ConnectionStatus(m_connectionType, m_remoteIp);
+            }
         }
 
-        public static async Task<ConnectionReturn> CheckConnectionAsync(string teamNumberS, TimeSpan timeout)
+        public bool Connected => m_remoteIp != null;
+
+
+        public RoboRioConnection(int teamNumber, TimeSpan sshTimeout)
         {
-            int teamNumber;
-            int.TryParse(teamNumberS, out teamNumber);
+            m_teamNumber = teamNumber;
+            m_sshTimeout = sshTimeout;
+        }
 
+        public void Dispose()
+        {
+            m_scpAdminClient?.Dispose();
+            m_sshUserClient?.Dispose();
+            m_scpUserClient?.Dispose();
+            m_scpAdminClient?.Dispose();
+        }
 
+        public static async Task<RoboRioConnection> StartConnectionTaskAsync(string teamNumberS)
+        {
+            int teamNumber;
+            int.TryParse(teamNumberS, out teamNumber);
             if (teamNumber < 0)
```

Not changed by your PR, but you should probably throw an exception here. This is bad user input, defaulting to 0 probably isn't going to give a good error.

---

There are other parts that check if it is 0, but that's a good point. I'll do that.

---

Will fix in a separate PR